### PR TITLE
Update of RAJA-based dense matrix to support runtime execution spaces

### PR DESCRIPTION
**Pull request by cnpetra**
Instrumentation of the RAJA dense matrix with the newly added execution space features. Related to issue #574

### CODE REVIEW

**Review by nychiang — APPROVED** [review]
LGTM. Just have some questions about `hiopVectorInt`. See the detailed comments.

File: src/LinAlg/hiopVectorRajaImpl.hpp [hunk 1]
```diff
@@ -287,7 +287,7 @@ void hiopVectorRaja<MEM,POL>::copyFrom(const double* local_array)
 
 /// @brief Copy from vec the elements specified by the indices in index_in_src.
 template<class MEM, class POL> 
-void hiopVectorRaja<MEM,POL>::copy_from(const hiopVector& vec, const hiopVectorInt& index_in_src)
+void hiopVectorRaja<MEM, POL>::copy_from(const hiopVector& vec, const hiopVectorInt& index_in_src)
 {
   const auto& v = dynamic_cast<const hiopVectorRaja<MEM, POL>&>(vec);
   const hiopVectorIntRaja& indexes = dynamic_cast<const hiopVectorIntRaja&>(index_in_src);
```
[the remark this answers is not in the document]
Comment (cnpetra) [reply src/LinAlg/hiopVectorRajaImpl.hpp:293]: yes, it is tracked by issue https://github.com/LLNL/hiop/issues/574 and I have in work.

File: tests/testMatrixSymSparse.cpp [hunk 2]
```diff
@@ -236,7 +236,8 @@ int main(int argc, char** argv)
 
     hiop::hiopVector* vec_m = hiop::LinearAlgebraFactory::create_vector(mem_space, M_global);
     hiop::hiopVector* vec_m_2 = hiop::LinearAlgebraFactory::create_vector(mem_space, M_global);
-    hiop::hiopMatrixRajaDense mxm_dense(2 * M_global, 2 * M_global, mem_space);
+    hiop::hiopMatrixDense* mxm_dense =
+      hiop::LinearAlgebraFactory::create_matrix_dense(mem_space, 2*M_global, 2*M_global);
```
[the remark this answers is not in the document]
Comment (cnpetra) [reply tests/testMatrixSymSparse.cpp:240]: works as designed via the constructor `ExecSpaceInfo::ExecSpaceInfo(string mem_space)`

File: src/LinAlg/hiopVectorRajaImpl.hpp [hunk 3]
```diff
@@ -1340,10 +1337,10 @@ void hiopVectorRaja<MEM,POL>::axdzpy(double alpha, const hiopVector& xvec, const
...
-void hiopVectorRaja<MEM,POL>::axdzpy_w_pattern(double alpha,
-                                               const hiopVector& xvec, 
-                                               const hiopVector& zvec,
-                                               const hiopVector& select)
+void hiopVectorRaja<MEM, POL>::axdzpy_w_pattern(double alpha,
+                                                const hiopVector& xvec, 
+                                                const hiopVector& zvec,
+                                                const hiopVector& select)
 {
   const hiopVectorRaja& x = dynamic_cast<const hiopVectorRaja<MEM, POL>&>(xvec);
```
[the remark this answers is not in the document]
Comment (cnpetra) [reply src/LinAlg/hiopVectorRajaImpl.hpp:1345]: just checked, it's only one (there wasn't one before, that diff shows new space)

File: src/LinAlg/hiopLinSolverSymDenseMagma.cpp [hunk 4]
```diff
@@ -1,6 +1,59 @@
...
+// 3. Also, reference herein to any specific commercial products, process, or services by
+// trade name, trademark, manufacturer or otherwise does not necessarily constitute or
+// imply its endorsement, recommendation, or favoring by the United States Government or
+// Lawrence Livermore National Security, LLC. The views and opinions of authors expressed
+// herein do not necessarily state or reflect those of the United States Government or
+// Lawrence Livermore National Security, LLC, and shall not be used for advertising or
+// product endorsement purposes.
+
+/**
+ * @file hiopLinSolverSymDenseMagma.hpp
```
[the remark this answers is not in the document]
Comment (cnpetra) [reply src/LinAlg/hiopLinSolverSymDenseMagma.cpp]: fixed

File: src/LinAlg/hiopMatrixDenseRaja.hpp [hunk 5]
```diff
@@ -277,11 +294,11 @@ class hiopMatrixRajaDense : public hiopMatrixDense
 
   //this is very private do not touch :)
   size_type max_rows_;
-
+
```
[the remark this answers is not in the document]
Comment (cnpetra) [reply src/LinAlg/hiopMatrixDenseRaja.hpp]: fixed

**Pull request merged**